pulsar-client/src/main/java/org/apache/pulsar/client/impl/PartitionedProducerImpl.java

```diff
@@ -113,25 +126,22 @@ private MessageRouter getMessageRouter() {
 
     @Override
     public String getProducerName() {
-        return producers.get(0).getProducerName();
+        return producers.get(firstPartitionIndex).getProducerName();
```

Comment:
This change breaks existing behavior. If it is not approved, I'll be changed to create the `partition-0` producer for backward compatibility.

Reply:
shouldn't be the same producerName for all the producers ?
why not computing the name in the constructor and cache it ?